Further improvements to Dark mode

mcornella opened the pull request:
I believe this PR fixes all Dark mode issues, so close issue #3798 once this is merged if you agree.

-----
1. LINKED ISSUES:
   Fix border issues in #3798:
   - [Pinned issues border](https://github.com/sindresorhus/refined-github/issues/3798#issuecomment-741568995).
   - [Divider line in grouped events in Activity list](https://github.com/sindresorhus/refined-github/issues/3798#issuecomment-741604005).

2. TEST URLS:
   - [Pinned issues example](https://github.com/sindresorhus/refined-github/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc).
   - [Activity list example](https://github.com/) (see while logged in).

3. SCREENSHOTS:

<table>
<tr>
	<td>Before
	<td>After
<tr>
	<td><img alt="pinned issues - before" src="https://user-images.githubusercontent.com/1441704/101666452-864d3900-3a4e-11eb-8b9d-97717301fcc3.png">
	<td><img alt="pinned issues - after" src="https://user-images.githubusercontent.com/1441704/101666047-03c47980-3a4e-11eb-9268-d04af97f2359.png">
</table>

<table>
<tr>
	<td>Before
	<td>After
<tr>
	<td><img alt="grouped events divider line - before" src="https://user-images.githubusercontent.com/1441704/101666533-a0871700-3a4e-11eb-8dd0-b8a802f5a07b.png">
	<td><img alt="grouped events divider line - after" src="https://user-images.githubusercontent.com/1441704/101666295-530aaa00-3a4e-11eb-8358-017f0f09ef73.png">
</table>

Since this PR uses the `--github-border-color` CSS variable already present in the codebase, this automatically fixes other border colors that use this variable. This includes table-input cell borders in comments:

<table>
<tr>
	<td>Before
	<td>After
<tr>
	<td><img alt="table-input cells - before" src="https://user-images.githubusercontent.com/1441704/101667248-636f5480-3a4f-11eb-82cc-dc215b78c3a5.png">
	<td><img alt="table-input cells - after" src="https://user-images.githubusercontent.com/1441704/101667138-4175d200-3a4f-11eb-8c1e-487cbaf32cc6.png">
</table>

This is a better approach than making `--github-border-color` a fallback like I did in the previous PR since it requires less changes.

---

## More fixes

1. Feature: `parse-backticks`. Use the `--color-markdown-code-bg` CSS variable to color the background in backticks parsed by the `parse-backticks` feature. To find the appropriate CSS variable I inspected the CSS of code elements in Markdown content (you can do it yourself with this very comment). [First reported here](https://github.com/sindresorhus/refined-github/issues/3798#issuecomment-742444029).

	<table>
	<tr>
		<td>Before
		<td>After
	<tr>
		<td><img alt="before" src="https://user-images.githubusercontent.com/1441704/102022936-4659bf00-3d8a-11eb-9e4a-911ab084c090.png">
		<td><img alt="after" src="https://user-images.githubusercontent.com/1441704/102022896-15798a00-3d8a-11eb-9aea-b5f6ddbec757.png">
	</table>

2. Feature: `split-issue-pr-search-results`. In a [GitHub search](https://github.com/search?q=asdfasdf&type=Issues), remove the background color in split Issues and PRs, then fix the color when hovering by using the `--color-menu-bg-active` CSS variable. I had to use `.menu-item` to add specificity to the selector, otherwise it could be done with `!important`.

	<table>
	<tr>
		<td>Before
		<td>After
	<tr>
		<td><img alt="before" src="https://user-images.githubusercontent.com/1441704/102022836-af8d0280-3d89-11eb-92a3-5c499d6aa41c.png">
		<td><img alt="after" src="https://user-images.githubusercontent.com/1441704/102022858-d9462980-3d89-11eb-9294-a67578ace46f.png">
	</table>

3. Feature: `resolve-conflicts`. When resolving a conflict, the first line number background uses the light color. We use `--color-auto-yellow-1` instead which is the coloring used by GitHub, and since `--github-faint-yellow` isn't used in any other place of the codebase, I removed it from `:root` and replaced its value.

	<table>
	<tr>
		<td>Before
		<td>After
	<tr>
		<td><img alt="before" src="https://user-images.githubusercontent.com/1441704/102023778-1f04f100-3d8e-11eb-9cef-903dd480eff4.png">
		<td><img alt="after" src="https://user-images.githubusercontent.com/1441704/102023796-347a1b00-3d8e-11eb-82ca-36423d233d34.png">
	</table>

4. Feature: many. Some features that use `--github-green` and `--github-red` appeared darker than the colors that GitHub uses now (`--color-text-success` and `--color-text-danger`).

	Red color:
	<table>
	<tr>
		<td>Before
		<td>After
	<tr>
		<td><img alt="before" src="https://user-images.githubusercontent.com/1441704/102025247-f6cdc000-3d96-11eb-826c-81f924872888.png">
		<td><img alt="after" src="https://user-images.githubusercontent.com/1441704/102025319-53c97600-3d97-11eb-8f62-a48fe62ba1ff.png">
	</table>
	
	Green color:
	<table>
	<tr>
		<td>Before
		<td>After
	<tr>
		<td><img alt="before" src="https://user-images.githubusercontent.com/1441704/102025256-0947f980-3d97-11eb-91ea-170f07a1a153.png">
		<td><img alt="after" src="https://user-images.githubusercontent.com/1441704/102025306-4c09d180-3d97-11eb-97b4-83114791d17f.png">
	</table>

5. Feature: `show-whitespace`. Fix the color of whitespace glyphs (fixes #3772).

	<table>
	<tr>
		<td>Before
		<td>After
	<tr>
		<td><img alt="before" src="https://user-images.githubusercontent.com/1441704/102115794-5336eb00-3e3c-11eb-8e5f-81f1bbc36bf0.png">
		<td><img alt="after" src="https://user-images.githubusercontent.com/1441704/102115891-72ce1380-3e3c-11eb-8cf1-8acdd208ea57.png">
	</table>

6. Feature: `table-input`. Soften the blue background color of cells.

	![image](https://user-images.githubusercontent.com/1441704/102227005-59cf6c00-3ee9-11eb-8704-23982c45e57e.png) ![image](https://user-images.githubusercontent.com/1441704/102227822-4b358480-3eea-11eb-8b07-2af71b8fb59b.png)

7. Feature: `conflict-marker`. Use a darker color for Dark mode (`#505050`).

    ![before](https://user-images.githubusercontent.com/1441704/102625121-4e787c80-4145-11eb-9a44-2fdae9f6980d.png) ![after](https://user-images.githubusercontent.com/1441704/102624980-14a77600-4145-11eb-98bc-1dcaf8ebe43f.png)

8. Feature: `emphasize-draft-pr-label`. Make it less bright on dark mode.

    ![before](https://user-images.githubusercontent.com/1441704/102625257-7cf65780-4145-11eb-8684-080fbd12f8d3.png) ![after](https://user-images.githubusercontent.com/1441704/102625374-afa05000-4145-11eb-960e-393e5db96b5e.png)

9. Feature: `highlight-collaborators-and-own-conversations`. Dim collaborator/own indicator border in PR lists

    ![before](https://user-images.githubusercontent.com/1441704/102625496-dfe7ee80-4145-11eb-97cb-7d0e7bf7cb51.png) ![after](https://user-images.githubusercontent.com/1441704/102625445-ccd51e80-4145-11eb-99d9-6518c591b200.png)

10. Feature: `reactions-avatars`. Fix reaction avatars in dark mode: (1) make the box-shadow color match the background, and (2) use a darker background color in avatars for when the image didn't load.

    ![before](https://user-images.githubusercontent.com/1441704/102624773-c98d6300-4144-11eb-80ea-d3f6ada4acc2.png) ![after](https://user-images.githubusercontent.com/1441704/102624898-f5104d80-4144-11eb-8961-9c7d3a06600f.png)

busches commented on source/features/reactions-avatars.css:
```diff
@@ -4,8 +4,8 @@
 button.reaction-summary-item { /* `button` excludes the "Add reaction" icon */
-	border-top: 1px solid var(--color-border-primary, #e1e4e8); /* Required when the second line is longer than the first line */
-	border-bottom: 1px solid var(--color-border-primary, #e1e4e8); /* Required when the first line is longer than the second line */
+	border-top: 1px solid var(--github-border-color); /* Required when the second line is longer than the first line */
```
Why are we losing the fallbacks here?

fregante replied:
They’re not part of the variable itself

busches replied:
So this old code was invalid `var(--color-border-primary, #e1e4e8)`? Man, CSS has somehow gotten itself even more complicated.

cheap-glitch replied:
> So this old code was invalid

No, it's just that now the fallback is defined directly on `--github-border-color`. DRY and all that.

busches replied:
Are we defining that? As enterprise appears to have zero variables when I checked the other day.

mcornella replied:
Yes, see the `refined-github.css` file diff.

busches replied:
Thanks, repo has been moving very fast lately and it's hard to keep up.

mcornella commented:
Added another fix for the background of parsed backticks text.

BTW, I found that commit lists in PR conversations [is no longer parsed correctly](https://github.com/sindresorhus/refined-github/pull/1986): 
![image](https://user-images.githubusercontent.com/1441704/101804845-52d3e280-3b12-11eb-9b6b-d144df4456c3.png)
I found that the `commit-message` class is no longer applied (I understand it was previously, based off the feature code?). The `js-commit` class is, though it's applied to the whole timeline item, but with this the feature works correctly again:
```patch
diff --git a/source/features/parse-backticks.tsx b/source/features/parse-backticks.tsx
index 4ee24d39..dbdf32ae 100644
--- a/source/features/parse-backticks.tsx
+++ b/source/features/parse-backticks.tsx
@@ -13,7 +13,7 @@ function init(): void {
         '.js-commits-list-item .mb-1, .js-commits-list-item pre', // `isCommitList` commit message and description
         '.Details[data-issue-and-pr-hovercards-enabled] .d-none a.link-gray-dark', // `isRepoRoot` commit message
         '.commit-title, .commit-desc', // `isCommit` commit message and description
-        '.commit-message', // `isPRConversation`, `isCompare`, `isReleasesOrTags` pushed commits
+        '.commit-message, .js-commit', // `isPRConversation`, `isCompare`, `isReleasesOrTags` pushed commits
         '.blame-commit-message', // `isBlame` commit message
         'a[id^="issue_"]', // `isConversationList` issue and PR title
         '.TimelineItem-body > del, .TimelineItem-body > ins', // `isIssue`, `isPRConversation` title edits
```

If I'm correct in my assessment, which I'm not sure, this is more an effect of the new GitHub UI version push and not an issue with dark mode, since light mode has the same markup. Should I push it here as well or make another commit?

yakov116 commented:
Separate PR

kidonng commented:
@mcornella Yeah this is what I mentioned in https://github.com/sindresorhus/refined-github/issues/3798#issuecomment-742561887:

> (And I remember seeing unparsed backticks some time ago, but that's another topic though)

mcornella commented:
@kidonng sorry, I had the branch ready yesterday. I pushed #3812, with an added fix for correctness (commit number 2).

fregante commented:
Can you react with 🚀 on each comment that this PR fixes in the main issue? So I mark them as resolved once I merge it

mcornella commented:
Done!

fregante commented:
Thanks! Are there any other hardcoded colors in the extension or should I merge this?

cheap-glitch commented:
> Thanks! Are there any other hardcoded colors in the extension or should I merge this?

Maybe #3772 could be also fixed in this PR?

kidonng commented:
> @kidonng sorry, I had the branch ready yesterday. I pushed #3812, with an added fix for correctness (commit number 2).
> 
> EDIT: you could still make a PR checking the rest of the selectors, I'm not familiar with them enough to know where in the GitHub UI they apply. You can even cherry-pick my commits and put them all into a single PR if you want.

@mcornella I will be busy for a couple of days, so go ahead if you find anything broken. I will comment on your PR if I have the time.

mcornella commented:
Some features that use `--github-green` and `--github-red` appear darker than the colors that GitHub uses now (`--color-text-success` and `--color-text-danger`). This doesn't constitute a bug but I wonder if I should add these in the definition of `--github-green` and `--github-red`?

<table>
<tr>
	<td>Before
	<td>After
<tr>
	<td><img alt="before" src="https://user-images.githubusercontent.com/1441704/102025247-f6cdc000-3d96-11eb-826c-81f924872888.png">
	<td><img alt="after" src="https://user-images.githubusercontent.com/1441704/102025319-53c97600-3d97-11eb-8f62-a48fe62ba1ff.png">
</table>

<table>
<tr>
	<td>Before
	<td>After
<tr>
	<td><img alt="before" src="https://user-images.githubusercontent.com/1441704/102025256-0947f980-3d97-11eb-91ea-170f07a1a153.png">
	<td><img alt="after" src="https://user-images.githubusercontent.com/1441704/102025306-4c09d180-3d97-11eb-97b4-83114791d17f.png">
</table>

mcornella commented:
Going through features that have color changes I stumbled upon `extend-status-labels.css`. Should this have been removed alongside `extend-status-labels.tsx` in b8614bdf?

mcornella commented:
Alright, I've checked the files in `source/` for color strings: rgb/a, hsl/a, #rgb and its variants, `color` and CSS color names. This is the result:

<details>
<summary>Fixed</summary>

```
source/features/clean-dashboard.css
source/features/clean-pinned-issues.css
source/features/parse-backticks.css
source/features/resolve-conflicts.css
source/features/show-whitespace.css
source/features/split-issue-pr-search-results.css
source/features/table-input.css
```

</details>

<details>
<summary>Checked and working</summary>

```
source/features/center-reactions-popup.css
source/features/clean-conversation-headers.css
source/features/clean-footer.css
source/features/clean-mergeability-box.css
source/features/conflict-marker.css
source/features/deep-reblame.css
source/features/emphasize-draft-pr-label.css
source/features/extend-diff-expander.css
source/features/extend-status-labels.css
source/features/faster-pr-diff-options.tsx
source/features/first-published-tag-for-merged-pr.tsx
source/features/git-checkout-pr.tsx
source/features/hidden-review-comments-indicator.css
source/features/hide-comments-faster.tsx
source/features/hide-readme-header.css
source/features/hide-useless-comments.css
source/features/highest-rated-comment.css
source/features/highest-rated-comment.tsx
source/features/highlight-collaborators-and-own-conversations.css
source/features/highlight-deleted-and-added-files-in-diffs.tsx
source/features/linkify-user-edit-history-popup.tsx
source/features/mark-merge-commits-in-list.css
source/features/mark-private-orgs.css
source/features/pinned-issues-update-time.tsx
source/features/pr-approvals-count.css
source/features/pr-commit-lines-changed.tsx
source/features/quick-fork-deletion.css
source/features/quick-mention.css
source/features/reactions-avatars.css
source/features/release-download-count.tsx
source/features/remove-label-faster.css
source/features/remove-label-faster.tsx
source/features/repo-age.tsx
source/features/show-associated-branch-prs-on-fork.tsx
source/features/submit-review-as-single-comment.tsx
source/features/suggest-commit-title-limit.css
source/features/useful-not-found-page.tsx
source/features/user-local-time.tsx
source/features/user-profile-follower-badge.tsx
source/features/wait-for-build.css
source/refined-github.css
```

</details>

Pending #3772 and https://github.com/sindresorhus/refined-github/pull/3806#issuecomment-744075848 this is almost ready.

----
These are the grep results sorted by result, in case anybody's interested:

<details>
<summary>Details</summary>

```
# Checked
source/features/center-reactions-popup.css:57:	color: #4078c0;
source/features/clean-conversation-headers.css:2:	background-color: #0366d6;
source/features/clean-conversation-headers.css:6:	color: #fff;
source/features/clean-footer.css:19:	color: #666;
source/features/clean-mergeability-box.css:14:	color: inherit;
source/features/clean-mergeability-box.css:20:	color: var(--github-red);
source/features/conflict-marker.css:3:	color: #c0c0c0;
source/features/deep-reblame.css:2:	color: #e40de48c;
source/features/deep-reblame.css:6:	color: #8000808c;
source/features/emphasize-draft-pr-label.css:2:	stroke: #586069;
source/features/emphasize-draft-pr-label.css:4:	color: #fff !important;
source/features/extend-diff-expander.css:4:	border-color: #0366d6;
source/features/extend-diff-expander.css:5:	color: #fff;
source/features/extend-diff-expander.css:6:	background: #0366d6;
source/features/extend-status-labels.css:8:	color: #fff !important;
source/features/faster-pr-diff-options.tsx:50:			className={`btn btn-sm btn-outline tooltipped tooltipped-s ${isHidingWhitespace ? 'bg-gray-light text-gray-light' : ''}`}
source/features/first-published-tag-for-merged-pr.tsx:37:			<TagIcon className="mx-1 text-gray-light v-align-middle"/>,
source/features/git-checkout-pr.tsx:36:			{remote && <p className="text-gray text-small my-1">{remoteType}</p>}
source/features/git-checkout-pr.tsx:83:			<p className="text-gray text-small">
source/features/hidden-review-comments-indicator.css:20:	color: transparent;
source/features/hidden-review-comments-indicator.css:32:	color: inherit;
source/features/hide-comments-faster.tsx:40:	hideCommentForm.classList.add('dropdown-menu', 'dropdown-menu-sw', 'text-gray-dark', 'show-more-popover', 'anim-scale-in');
source/features/hide-readme-header.css:10:	background-color: transparent !important;
source/features/hide-useless-comments.css:4:	color: #586069;
source/features/hide-useless-comments.css:8:	border-color: var(--github-red);
source/features/hide-useless-comments.css:8:	border-color: var(--github-red);
source/features/highest-rated-comment.css:18:	--color-auto-gray-1: var(--color-auto-green-1);
source/features/highest-rated-comment.css:18:	--color-auto-gray-1: var(--color-auto-green-1);
source/features/highest-rated-comment.css:19:	--color-auto-blue-4: var(--color-auto-green-4);
source/features/highest-rated-comment.css:19:	--color-auto-blue-4: var(--color-auto-green-4);
source/features/highest-rated-comment.css:2:	border: 2px solid #2cbe4e !important;
source/features/highest-rated-comment.css:20:	--color-auto-blue-2: var(--color-auto-green-2);
source/features/highest-rated-comment.css:20:	--color-auto-blue-2: var(--color-auto-green-2);
source/features/highest-rated-comment.css:24:	--color-bg-tertiary: var(--color-auto-green-1);
source/features/highest-rated-comment.css:24:	--color-bg-tertiary: var(--color-auto-green-1);
source/features/highest-rated-comment.tsx:53:	avatar.append(<CheckIcon width={24} height={32} className="mt-4 text-green"/>);
source/features/highest-rated-comment.tsx:58:			className="d-flex flex-items-center text-green mr-1 tooltipped tooltipped-n"
source/features/highest-rated-comment.tsx:80:			<a href={hash} className="no-underline rounded-1 rgh-highest-rated-comment timeline-comment bg-gray px-2 d-flex flex-items-center">
source/features/highest-rated-comment.tsx:86:				<span className="text-gray timeline-comment-header-text">
source/features/highlight-collaborators-and-own-conversations.css:2:	border: 1px solid #c0d3eb;
source/features/highlight-deleted-and-added-files-in-diffs.tsx:46:				icon.classList.add('text-green');
source/features/highlight-deleted-and-added-files-in-diffs.tsx:48:				icon.classList.add('text-red');
source/features/linkify-user-edit-history-popup.tsx:15:			wrap(avatar.nextElementSibling!, <a className="link-gray-dark" href={`/${userName}`}/>);
source/features/mark-merge-commits-in-list.css:6:	color: #4078c0;
source/features/mark-private-orgs.css:10:	fill: #333;
source/features/mark-private-orgs.css:11:	stroke: #fff;
source/features/pinned-issues-update-time.tsx:48:			<span className="text-gray d-inline-block">
source/features/pr-approvals-count.css:10:	color: var(--github-green);
source/features/pr-approvals-count.css:15:	color: var(--github-red) !important;
source/features/pr-commit-lines-changed.tsx:34:			<span className="text-green">+{additions}</span>{' '}
source/features/pr-commit-lines-changed.tsx:35:			<span className="text-red">−{deletions}</span>{' '}
source/features/quick-fork-deletion.css:8:	background-color: var(--color-scale-red-6);
source/features/quick-fork-deletion.css:8:	background-color: var(--color-scale-red-6);
source/features/quick-mention.css:12:	color: inherit;
source/features/reactions-avatars.css:17:	--background: #f2f8fa;
source/features/reactions-avatars.css:27:	background: #efefef; /* Placeholder before the images load */
source/features/reactions-avatars.css:28:	box-shadow: 0 0 0 2px var(--background, #fff);
source/features/reactions-avatars.css:47:	background-color: var(--background);
source/features/reactions-avatars.css:7:	border-top: 1px solid var(--github-border-color); /* Required when the second line is longer than the first line */
source/features/reactions-avatars.css:8:	border-bottom: 1px solid var(--github-border-color); /* Required when the first line is longer than the second line */
source/features/release-download-count.tsx:72:					const classes = 'rgh-release-download-count mr-2 text-gray' + (index === 0 ? ' text-bold' : '');
source/features/remove-label-faster.css:19:	background-color: currentColor;
source/features/remove-label-faster.css:20:	fill: var(--rgh-remove-label-faster-color);
source/features/remove-label-faster.css:5:	color: inherit !important;
source/features/remove-label-faster.tsx:51:						'--rgh-remove-label-faster-color': label.style.backgroundColor
source/features/repo-age.tsx:120:		<li className="text-gray" title={`First commit dated ${dateFormatter.format(birthday)}`}>
source/features/show-associated-branch-prs-on-fork.tsx:62:	Open: '--green',
source/features/show-associated-branch-prs-on-fork.tsx:63:	Closed: '--red',
source/features/show-associated-branch-prs-on-fork.tsx:64:	Merged: '--purple',
source/features/submit-review-as-single-comment.tsx:15:	<button className="rgh-submit-single btn-link ml-2 text-gray text-small" type="button">Send now</button>
source/features/suggest-commit-title-limit.css:3:	border-color: var(--github-red);
source/features/suggest-commit-title-limit.css:4:	background-color: rgb(203 36 49 / 3%);
source/features/suggest-commit-title-limit.css:9:	box-shadow: inset 0 1px 2px rgb(35 27 27 / 7.5%), 0 0 0 0.2em rgb(203 36 49 / 30%);
source/features/useful-not-found-page.tsx:21:	return <del style={{color: '#6a737d'}}>{text}</del>;
source/features/user-local-time.tsx:110:			<div className="rgh-local-user-time mt-2 text-gray text-small">
source/features/user-profile-follower-badge.tsx:26:			newProfileElement.after(<span className="text-gray"> · Follows you</span>);
source/features/wait-for-build.css:4:	color: #808080;
source/refined-github.css:2:    --github-green: var(--color-text-success, #28a745);
source/refined-github.css:3:    --github-red: var(--color-text-danger, #cb2431);
source/refined-github.css:4:    --github-gray-background: #fafbfc;
source/refined-github.css:5:    --github-gray-text: #6a737d;
source/refined-github.css:6:    --github-border-color: var(--color-border-primary, #e1e4e8);
source/refined-github.css:76:   color: #6a737d;

# Fixed
source/features/clean-dashboard.css:19:	background-color: var(--color-bg-canvas, #fff);
source/features/clean-dashboard.css:20:	border: 1px solid var(--github-border-color);
source/features/clean-dashboard.css:66:	border-top: 1px solid var(--github-border-color);
source/features/clean-pinned-issues.css:14:		box-shadow: 0 0 0 2px var(--github-border-color);
source/features/clean-pinned-issues.css:22:		border-color: var(--github-border-color) !important;
source/features/parse-backticks.css:3:	background-color: var(--color-markdown-code-bg, rgb(27 31 35 / 5%));
source/features/resolve-conflicts.css:3:	background-color: var(--github-faint-yellow);
source/features/show-whitespace.css:15:	background-image: url('data:image/svg+xml,%3Csvg preserveAspectRatio="xMinYMid meet" viewBox="0 0 12 24" xmlns="http://www.w3.org/2000/svg"%3E%3Cpath d="M9.5 10.44L6.62 8.12L7.32 7.26L12.04 11V11.44L7.28 14.9L6.62 13.9L9.48 11.78H0V10.44H9.5Z" fill="rgba(36,41,46,25%)"/%3E%3C/svg%3E');
source/features/show-whitespace.css:20:	background-image: url('data:image/svg+xml,%3Csvg preserveAspectRatio="xMinYMid meet" viewBox="0 0 12 24" xmlns="http://www.w3.org/2000/svg"%3E%3Cpath d="M4.5 11C4.5 10.1716 5.17157 9.5 6 9.5C6.82843 9.5 7.5 10.1716 7.5 11C7.5 11.8284 6.82843 12.5 6 12.5C5.17157 12.5 4.5 11.8284 4.5 11Z" fill="rgba(36,41,46,25%)"/%3E%3C/svg%3E');
source/features/split-issue-pr-search-results.css:2:	background-color: #f6f8fa;
source/features/table-input.css:16:	border: 2px solid var(--github-border-color);
source/features/table-input.css:21:	border-color: #79b8ff;
source/features/table-input.css:22:	background-color: #dbedff;
```

</details>

And I used these two commands:
```
grep -n -rE '(rgb|rgba|hsl|hsla)\(|#([0-9a-f]{3}){1,2}([0-9a-f]{2}|\b)|color' source/ | sort -u
grep -n -rE '\baliceblue\b|\bantiquewhite\b|\baqua\b|\baquamarine\b|\bazure\b|\bbeige\b|\bbisque\b|\bblack\b|\bblanchedalmond\b|\bblue\b|\bblueviolet\b|\bbrown\b|\bburlywood\b|\bcadetblue\b|\bchartreuse\b|\bchocolate\b|\bcoral\b|\bcornflowerblue\b|\bcornsilk\b|\bcrimson\b|\bcyan\b|\bdarkblue\b|\bdarkcyan\b|\bdarkgoldenrod\b|\bdarkgray\b|\bdarkgrey\b|\bdarkgreen\b|\bdarkkhaki\b|\bdarkmagenta\b|\bdarkolivegreen\b|\bdarkorange\b|\bdarkorchid\b|\bdarkred\b|\bdarksalmon\b|\bdarkseagreen\b|\bdarkslateblue\b|\bdarkslategray\b|\bdarkslategrey\b|\bdarkturquoise\b|\bdarkviolet\b|\bdeeppink\b|\bdeepskyblue\b|\bdimgray\b|\bdimgrey\b|\bdodgerblue\b|\bfirebrick\b|\bfloralwhite\b|\bforestgreen\b|\bfuchsia\b|\bgainsboro\b|\bghostwhite\b|\bgold\b|\bgoldenrod\b|\bgray\b|\bgrey\b|\bgreen\b|\bgreenyellow\b|\bhoneydew\b|\bhotpink\b|\bindianred\b|\bindigo\b|\bivory\b|\bkhaki\b|\blavender\b|\blavenderblush\b|\blawngreen\b|\blemonchiffon\b|\blightblue\b|\blightcoral\b|\blightcyan\b|\blightgoldenrodyellow\b|\blightgray\b|\blightgrey\b|\blightgreen\b|\blightpink\b|\blightsalmon\b|\blightseagreen\b|\blightskyblue\b|\blightslategray\b|\blightslategrey\b|\blightsteelblue\b|\blightyellow\b|\blime\b|\blimegreen\b|\blinen\b|\bmagenta\b|\bmaroon\b|\bmediumaquamarine\b|\bmediumblue\b|\bmediumorchid\b|\bmediumpurple\b|\bmediumseagreen\b|\bmediumslateblue\b|\bmediumspringgreen\b|\bmediumturquoise\b|\bmediumvioletred\b|\bmidnightblue\b|\bmintcream\b|\bmistyrose\b|\bmoccasin\b|\bnavajowhite\b|\bnavy\b|\boldlace\b|\bolive\b|\bolivedrab\b|\borange\b|\borangered\b|\borchid\b|\bpalegoldenrod\b|\bpalegreen\b|\bpaleturquoise\b|\bpalevioletred\b|\bpapayawhip\b|\bpeachpuff\b|\bperu\b|\bpink\b|\bplum\b|\bpowderblue\b|\bpurple\b|\brebeccapurple\b|\bred\b|\brosybrown\b|\broyalblue\b|\bsaddlebrown\b|\bsalmon\b|\bsandybrown\b|\bseagreen\b|\bseashell\b|\bsienna\b|\bsilver\b|\bskyblue\b|\bslateblue\b|\bslategray\b|\bslategrey\b|\bsnow\b|\bspringgreen\b|\bsteelblue\b|\btan\b|\bteal\b|\bthistle\b|\btomato\b|\bturquoise\b|\bviolet\b|\bwheat\b|\bwhite\b|\bwhitesmoke\b|\byellow\b|\byellowgreen\b' source/ | sort -u
```

fregante commented:
> Going through features that have color changes I stumbled upon `extend-status-labels.css`. Should this have been removed alongside `extend-status-labels.tsx` in [b8614bd](https://github.com/sindresorhus/refined-github/commit/b8614bdfee788989f5032b0f83afdd0af012904d)?

Can be done in a future lint PR.

fregante commented on source/refined-github.css:
```diff
@@ -1,10 +1,9 @@
+	--github-green: var(--color-text-success, #28a745);
+	--github-red: var(--color-text-danger, #cb2431);
 	--github-gray-background: #fafbfc;
 	--github-gray-text: #6a737d;
```
These 2 aren't used anywhere, can be dropped

fregante commented on source/refined-github.css:
```diff
@@ -1,10 +1,9 @@
 :root {
-	--github-green: #28a745;
-	--github-red: #cb2431;
+	--github-green: var(--color-text-success, #28a745);
```
Let's add a comment here mentioning that these variables only exist for GHE and should be removed in a few months.

fregante commented:
After the 2 comments are fixed, let's merge this, together with https://github.com/sindresorhus/refined-github/pull/3812, and release a new hotfix version.

fregante commented:
<img width="388" alt="" src="https://user-images.githubusercontent.com/1402241/102157124-ed9f2a80-3e4c-11eb-9925-50e7acc094ca.png">

The conflicts icon should also be dimmer. This is currently visible in our PR list

fregante commented:
This is still too bright, comparatively:

<img width="202" alt="Screen Shot 2020-12-14 at 20 52 51" src="https://user-images.githubusercontent.com/1402241/102158019-871b0c00-3e4e-11eb-91e2-7a1a67538471.png">
<img width="170" alt="Screen Shot 2020-12-14 at 20 52 57" src="https://user-images.githubusercontent.com/1402241/102158021-87b3a280-3e4e-11eb-87f0-69d7ec8b7648.png">

mcornella commented:
> This is still too bright:

1. Current
2. Using `--color-scale-blue-1` (same value for light mode)
3. Using `--color-auto-blue-1` (same value for light mode)

![image](https://user-images.githubusercontent.com/1441704/102227005-59cf6c00-3ee9-11eb-8704-23982c45e57e.png) ![image](https://user-images.githubusercontent.com/1441704/102227000-589e3f00-3ee9-11eb-8887-c19178a29098.png) ![image](https://user-images.githubusercontent.com/1441704/102227552-fd208100-3ee9-11eb-920a-6a725373d0d5.png)

Which one do you prefer?

fregante commented:
4 looks good. 

In a separate PR (maybe with #3823), I wonder how it would look if we dropped the spacing between each cell, so it looks like an actual table

fregante commented:
No need to force push because it makes it hard to know what changed

mcornella commented:
The only commit I modified was https://github.com/sindresorhus/refined-github/pull/3806/commits/0ea87d7faa5e0b5455aa8947bb5f5e318def40df where I only removed the extra CSS variables and added a comment. I've yet to commit https://github.com/sindresorhus/refined-github/pull/3806#issuecomment-745406296, let me know what you think about each proposal and I'll make the commits.

fregante commented:
Looks good

fregante commented on source/features/reactions-avatars.css:
```diff
@@ -25,10 +29,14 @@ button.reaction-summary-item { /* `button` excludes the "Add reaction" icon */
 	margin-left: -0.5em;
 	vertical-align: middle;
 	background: #efefef; /* Placeholder before the images load */
```
Can you use an existing variable here instead?

mcornella replied:
There's no variable in https://primer.style/primitives/ that corresponds with `#efefef` for light mode. This is using `--color-bg-canvas-inset`:
![image](https://user-images.githubusercontent.com/1441704/102628746-7a4a3100-414a-11eb-800c-c1c23c2cd636.png)
![image](https://user-images.githubusercontent.com/1441704/102628752-7b7b5e00-414a-11eb-9e26-95105deb97b2.png)

fregante commented on source/features/conflict-marker.css:
```diff
@@ -2,3 +2,7 @@
 }
+
+[data-color-mode='dark'] .rgh-conflict-marker svg {
+	color: #505050;
```
This could use a variable too, the color doesn't need to be exactly what it was. If it's near enough as long as it looks good

fregante commented on source/features/emphasize-draft-pr-label.css:
```diff
@@ -1,7 +1,7 @@
+	stroke: var(--color-text-secondary, #586069);
 	stroke-width: 1.2px;
-	color: #fff !important;
+	color: var(--color-bg-canvas, #fff) !important;
```
I think this is more appropriate, without using `important`, I don't know if it works

```suggestion
	fill: var(--color-bg-canvas, #fff);
```

fregante reviewed (commented):
After this it can probably be merged

mcornella commented:
Sorry, I saw the last comments. Thanks for merging!

fregante commented:
Yeah I don’t like ever-increasing long-living PRs, especially for bug fixes, it’s best to limit the scope and merge often. The remaining comments and further improvements can be done in a new PR

fregante commented:
It looks like the color doesn't work for _your own_ reactions. See here:

<img width="428" alt="Screen Shot 3" src="https://user-images.githubusercontent.com/1402241/104117473-24028500-52e7-11eb-9d87-b0ab32c1c6c0.png">

mcornella commented:
I specifically targeted this in https://github.com/sindresorhus/refined-github/pull/3806/files#diff-da8c7be7c768eb7bc8b49010c3f06209916a1d2282d2069d10646b13731582fdR20, has something changed?

shinenelson commented:
the background of the emoji too is white for me because 
```css
.reaction-summary-item img {
  background-color: var(--background)
}
```

![image](https://user-images.githubusercontent.com/4771718/104836977-a2f35100-58d7-11eb-8b96-94535267bd35.png)

apparently, the specifically dark mode targeted fix has not worked for me either.

mcornella commented:
Can you post a screenshot of the Styles panel of the element in devtools? Also, is `data-color-mode=dark` still an attribute of the `<html>` tag?

shinenelson commented:
> Also, is data-color-mode=dark still an attribute of the <html> tag?

Ah, damn, that was the issue for me - my `data-color-mode` was `auto` because I have system-wide dark mode enabled and Firefox seems to honor the system setting by default. I manually changed the value of the attribute to `dark` in DevTools and it worked as expected. 

Personally, I do not want to set dark mode explicitly. I would rather let my global setting be honored wherever possible. So I guess I will have to live with this :man_facepalming:

shinenelson commented:
can we maybe change the CSS rule to use `prefers-color-scheme: dark` rather than use the `data-color-mode`? I am not sure how this would affect the rest of the extension though.

mcornella commented:
That's right I hadn't thought about this case. All the CSS selectors that use `[data-color-mode=dark]` should also target `@media (prefers-color-scheme: dark) [data-color-mode=auto]`. 

There are not many that use this, you can quickly see them in the Files tab of this PR. Unfortunately I don't have time for this, but it should be a quick fix for anyone trying.
